lib/arrays.md

```diff
@@ -5,44 +5,91 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
-Replace me with your response...
+A list a cat's favorite things. All 12 months. The alphabet. Dinosaur types. Prime numbers. 
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
+Same concept as in strings. the first element in an array is in the 0 position, second is in the 1 position and so on.
+
+    prime = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]
+    prime[6]
+     will return 17
+    prime[8]
+     will return 23
+    prime[99]
+     will return 'nil' because it does not exist
 
-Replace me with your response...
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+Same concept as in strings. Last element in array will be in the -1 position, second to last will be in the -2 position, and so on.
 
+    prime[-6]
+     will return 29
+    
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+    prime.each do |x|
+     puts x+=1
+    end
+    
+This would add 1 to each element in the array
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+You can insert something in a specific location:
+
+    prime.insert(-1, 53)
+     inserts 53 in the -1 position of the array
+    prime << 53
+     puts 53 at the last positon of the array
+    prime.push(53).push(59)
+     pushes the new elements to the end of the array. Able to chain multiple pushes together
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
+The .map method creates a new array based on the original array, but changes the elements based on your block.
 
-Replace me with your response...
+    arr = ["Laura", "Fiona", "Tori"]
+    arr.map! { |element|
+       if(element == "Fiona")
+           "Florence"
+       else
+           element
+       end
+    }
+
+    => ["Laura", "Florence", "Tori"]
 
 # What do the methods `push`, `pop`, `shift`, and `unshift` do?
 
-Replace me with your response...
+Push adds one or more new elements to the end of an array.
+Pop removes the last element of an array and returns it.
+Shift removes the first element of an array and returns it.
+Unshift will add a new item to the beginning of an array.
 
 # How do you determine how many elements are in an Array?
 
-Replace me with your response...
+Use the count operation:
+
+    prime.count
+        => 17
 
 # How would you reverse the order of elements in an Array?
 
-Replace me with your response...
+    arr = ["Laura", "Florence", "Tori"]
+    arr.reverse_each { |x| puts x }
```

Comment:
`reverse_each` loops over all of the elements in an array in reversed order, but it doesn't actually reverse the elements themselves. Take the following code:

```ruby
arr = ["Laura", "Florence", "Tori"]
reversed_order = arr.YOUR_ANSWER_HERE
```

There should be a quick and easy way to get the reversed order and stored it as a separate array.